### PR TITLE
Suppress a lot of warnings

### PR DESCRIPTION
**Pull request by okkez**
This is the first step to wipe out warnings.

### CODE REVIEW

**Comment by okkez** [comment]
Some warnings may conflict with other PRs, so I leave these warnings.

File: lib/fluent/parser.rb [hunk 1]
```diff
@@ -155,18 +155,10 @@ def parse(text)
             when @time_key
               time = @mutex.synchronize { @time_parser.parse(value) }
               if @keep_time_key
-                record[name] = if @type_converters.nil?
-                                 value
-                               else
-                                 convert_type(name, value)
-                               end
+                record[name] = convert_type(name, value)
```
Comment (repeatedly) [inline lib/fluent/parser.rb]: Calling `convert_type` heavily is high cost.
So we should not call `convert_type` when user doesn't specify `types` parameter.

File: lib/fluent/mixin.rb [hunk 2]
```diff
@@ -133,10 +133,12 @@ module HandleTagNameMixin
     attr_accessor :remove_tag_prefix, :remove_tag_suffix, :add_tag_prefix, :add_tag_suffix
     def configure(conf)
       super
+      @remove_tag_prefix = nil
```
Comment (repeatedly) [inline lib/fluent/mixin.rb]: How about `@remove_tag_prefix = conf.has_key?('remove_tag_prefix') ? Regexp.new('^' + Regexp.escape(conf['remove_tag_prefix'])) : nil`?

File: lib/fluent/mixin.rb [hunk 3]
```diff
@@ -133,10 +133,12 @@ module HandleTagNameMixin
...
       if remove_tag_prefix = conf['remove_tag_prefix']
         @remove_tag_prefix = Regexp.new('^' + Regexp.escape(remove_tag_prefix))
       end
 
+      @remove_tag_suffix = nil
```
Comment (repeatedly) [inline lib/fluent/mixin.rb]: ditto

**Comment by okkez** [comment]
I have fixed and pushed.
How about this?

**Comment by tagomoris** [comment]
LGTM. (Squash required after all reviewer comments.)

**Comment by sonots** [comment]
Is it possible to integrate ruby warning check into CI? Otherwise, I think same things happen again in the future.

**Comment by sonots** [comment]
Good job anyway, this allows plugin developers to use ruby -w for their own plugins. Before, fluentd itself showed lots of warnings, so ruby -w for plugins did not work effectively.

**Comment by tagomoris** [comment]
@sonots what is "ruby warning check" you mentioned?

**Comment by sonots** [comment]
Checking whether ruby -w tells warnings or not

**Comment by tagomoris** [comment]
It's impossible to remove all warnings immediately (because there are some/many warnings from libraries which fluentd depends on).
IMO, checking `ruby -w` outputs looks good idea at a point in future, but we cannot add it right now.

**Comment by okkez** [comment]
@tagomoris squashed!

**Comment by tagomoris** [comment]
@okkez Thanks! I'll merge this after Travis green (AppVeyor looks in trouble now).

**Comment by tagomoris** [comment]
Merged. Thank you!

**Pull request merged**